crates/router/src/types/storage/refund.rs

```diff
@@ -69,4 +86,73 @@ impl RefundDbExt for Refund {
             .change_context(errors::DatabaseError::NotFound)
             .attach_printable_lazy(|| "Error filtering records by predicate")
     }
+
+    async fn filter_by_meta_constraints(
+        conn: &PgPooledConn,
+        merchant_id: &str,
+        refund_list_details: &api_models::refunds::RefundListRequest,
+        limit: i64,
+    ) -> CustomResult<api_models::refunds::RefundListMetaData, errors::DatabaseError> {
+        let start_time = refund_list_details
+            .time_range
+            .map(|t| t.start_time)
+            .unwrap_or(time::macros::datetime!(2022-06-20 00:00:00));
+        let end_time = refund_list_details
+            .time_range
+            .and_then(|t| t.end_time)
+            .unwrap_or_else(common_utils::date_time::now);
+
+        let filter = <Self as HasTable>::table()
+            .filter(dsl::merchant_id.eq(merchant_id.to_owned()))
+            .order(dsl::modified_at.desc())
+            .limit(limit)
+            .filter(dsl::created_at.ge(start_time))
+            .filter(dsl::created_at.le(end_time));
+
+        let filter_connector: Vec<String> = filter
+            .clone()
+            .select(dsl::connector)
+            .distinct()
+            .order_by(dsl::connector.asc())
+            .get_results_async(conn)
+            .await
+            .into_report()
+            .change_context(errors::DatabaseError::NotFound)
+            .attach_printable_lazy(|| "Error filtering records by predicate")?;
```

Comment:
Neither of this and the following 2 errors should be `DatabaseError::NotFound`. In case no records are found, you'll get an empty `Vec`. So the only time you do get an error is for other database errors, which would be an application internal server error.

Change this to `DatabaseError::Others`, and add a meaningful message to each error with the aid of `attach_printable()` or `attach_printable_lazy()`.